### ISSUE TITLE
questions about run_conv_net.py

### ISSUE THREAD

**Issue by WillieMaddox** [blackecho/Deep-Learning-TensorFlow#29]
Hello again,

I am trying to run run_conv_net.py. Here is my command:

```
time python command_line/run_conv_net.py 
--main_dir covnet-mnist-models 
--model_name covnet-mnist 
--layers conv2d-5-5-32-1,maxpool-2,conv2d-5-5-64-1,maxpool-2,full-1024,softmax 
--batch_size 50 
--verbose 1 
--learning_rate 1e-4 
--opt adam 
--dropout 0.5
```
And here is my output:

```
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcublas.so.7.5 locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcufft.so.7.5 locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcurand.so.7.5 locally
Extracting MNIST_data/train-images-idx3-ubyte.gz
Extracting MNIST_data/train-labels-idx1-ubyte.gz
Extracting MNIST_data/t10k-images-idx3-ubyte.gz
Extracting MNIST_data/t10k-labels-idx1-ubyte.gz
Creating /home/maddoxw/.yadlt/stored_models/covnet-mnist-models directory to save/restore models
Creating /home/maddoxw/.yadlt/data/covnet-mnist-models directory to save model generated data
Creating /home/maddoxw/.yadlt/logs/covnet-mnist-models directory to save tensorboard logs
Start Convolutional Network training...
Building Convolutional layer with 1 input channels and 32 5x5 filters with stride 1
Building Max Pooling layer with size 2
Building Convolutional layer with 32 input channels and 64 5x5 filters with stride 1
Building Max Pooling layer with size 2
Building fully connected layer with 3136 in unitsand 1024 out units
Building softmax layer with 1024 in units and10 out units
I tensorflow/core/common_runtime/gpu/gpu_init.cc:102] Found device 0 with properties: 
name: GeForce GTX TITAN X
major: 5 minor: 2 memoryClockRate (GHz) 1.2155
pciBusID 0000:03:00.0
Total memory: 12.00GiB
Free memory: 10.34GiB
I tensorflow/core/common_runtime/gpu/gpu_init.cc:126] DMA: 0 
I tensorflow/core/common_runtime/gpu/gpu_init.cc:136] 0:   Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:843] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX TITAN X, pci bus id: 0000:03:00.0)
Tensorboard logs dir for this run is /home/maddoxw/.yadlt/logs/covnet-mnist-models/run4
Accuracy at step 0: 0.0958
Accuracy at step 1: 0.0958
Accuracy at step 2: 0.0958
Accuracy at step 3: 0.0958
Accuracy at step 4: 0.0958
Accuracy at step 5: 0.0958
Accuracy at step 6: 0.0958
Accuracy at step 7: 0.0958
Accuracy at step 8: 0.0958
Accuracy at step 9: 0.0958
I tensorflow/core/common_runtime/gpu/gpu_device.cc:843] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX TITAN X, pci bus id: 0000:03:00.0)
Test set accuracy: 0.0979999899864

real	2m25.073s
user	2m53.696s
sys 	0m27.416s
```
However the accuracy isn't changing. Should it be?

[This paper](http://yann.lecun.com/exdb/publis/pdf/jarrett-iccv-09.pdf) uses the same layers architecture as the one found in your readthedocs example, so I don't think that is the problem.

Also, here are the saved results from the run:

```
maddoxw@wmdevbox:~$ ls -la /home/maddoxw/.yadlt/stored_models/covnet-mnist-models
total 13M
drwxrwxr-x 2 maddoxw maddoxw 4.0K Sep 14 11:14 ./
drwxrwxr-x 8 maddoxw maddoxw 4.0K Sep  6 14:39 ../
-rw-rw-r-- 1 maddoxw maddoxw  191 Sep 14 11:14 checkpoint
-rw-rw-r-- 1 maddoxw maddoxw  13M Sep 14 11:14 covnet-mnist
-rw-rw-r-- 1 maddoxw maddoxw  56K Sep 14 11:14 covnet-mnist.meta
```
Can I load covnet-mnist directly into tensorflow?

```
maddoxw@wmdevbox:~$ ls -la /home/maddoxw/.yadlt/data/covnet-mnist-models
total 8.0K
drwxrwxr-x 2 maddoxw maddoxw 4.0K Sep  1 18:24 ./
drwxrwxr-x 8 maddoxw maddoxw 4.0K Sep  6 14:39 ../
```
Should this directory be empty?

```
maddoxw@wmdevbox:~$ ls -la /home/maddoxw/.yadlt/logs/covnet-mnist-models/run4
total 88K
drwxr-xr-x 2 maddoxw maddoxw 4.0K Sep 14 10:56 ./
drwxrwxr-x 7 maddoxw maddoxw 4.0K Sep 14 11:12 ../
-rw-rw-r-- 1 maddoxw maddoxw  77K Sep 14 10:58 events.out.tfevents.1473868563.wmdevbox
```
Could you provide a few pointers on how to use Tensorboard to read the binary log files?

Thanks for sharing.

**Comment by WillieMaddox** [comment]
Last question had a easy answer.
```
tensorboard /home/maddoxw/.yadlt/logs/covnet-mnist-models/run4/
```

**Comment by WillieMaddox** [comment]
Update: 

Still no joy on getting `run_conv_net` to produce meaningful results.  I ran [this CNN.py script](https://github.com/jorditorresBCN/LibroTensorFlow/blob/master/CNN.py) on the same mnist data and it seems to be working fine.  I spent some time comparing this script to `run_conv_net` and its dependencies.  I wasn't able to verify that both implementations of calculating cross_entropy were the same, but other than that, the programming logic is identical.

Thoughts,
